### PR TITLE
Diff across message sizes

### LINKED ISSUE #67: Calculate diff doesn't look at messageSize
Calculate diff needs to take into account the message size.

### PR DESCRIPTION
**Pull request by jtaleric**
Isolate to single stream for the diff, add different message sizes.

Closes #67

### CODE REVIEW

File: cmd/k8s-netperf/k8s-netperf.go [hunk 1]
```diff
@@ -239,16 +239,17 @@ var rootCmd = &cobra.Command{
...
-			diff, err := result.TCPThroughputDiff(sr)
-			if err != nil {
-				fmt.Println("Unable to calculate difference between HostNetwork and PodNetwork")
-				retCode = 1 //nolint:all
-			}
-			if diff < tcpt {
-				retCode = 0 //nolint:all
+			diffs, err := result.TCPThroughputDiff(&sr)
+			for _, diff := range diffs {
+				if err != nil {
```
Comment (vishnuchalla) [inline cmd/k8s-netperf/k8s-netperf.go:244]: Do we need to consider having this error check outside the for loop?

File: pkg/results/result.go [hunk 2]
```diff
@@ -95,26 +95,77 @@ func CheckHostResults(s ScenarioResults) bool {
...
-	hostPerf := 0.0
-	podPerf := 0.0
+	diffRes := []DiffData{}
 	for t := range s.Results {
 		if !s.Results[t].Service {
+			hostPerf := 0.0
+			podPerf := 0.0
+			diff := DiffData{}
 			if s.Results[t].HostNetwork {
 				if s.Results[t].Profile == "TCP_STREAM" {
```
Comment (vishnuchalla) [inline pkg/results/result.go:120]: can we invert the if check here?
```
check of TCP_stream first and
then
if hostnework{}
else{}
```

File: cmd/k8s-netperf/k8s-netperf.go [hunk 3]
```diff
@@ -239,16 +239,17 @@ var rootCmd = &cobra.Command{
...
+					fmt.Println("Unable to calculate difference between HostNetwork and PodNetwork")
+					retCode = 1 //nolint:all
+				}
+				if tcpt < diff.Result {
+					fmt.Printf("😥 TCP Single Stream (Message Size : %d) percent difference when comparing hostNetwork to podNetwork is greater than %.1f percent (%.1f percent)\r\n", diff.MessageSize, tcpt, diff.Result)
+					retCode = 1 //nolint:all
```
[the remark this answers is not in the document]
Comment (jtaleric) [reply cmd/k8s-netperf/k8s-netperf.go:250]: I was following the `nolint:all` that was here originally.

**Review by vishnuchalla — APPROVED** [review]
LGTM!

**Pull request merged**